### ISSUE TITLE
`Fallback to unknown` generates invalid type name

### ISSUE THREAD

**Issue by hugop95** [antfu/eslint-typegen#10]
### Describe the bug

Hi,

When an invalid JSON schema is provided for a rule, `eslint-typegen` falls back to `unknown` for the created type. That generated type may cause compilation to fail:

This issue was detected from https://github.com/azat-io/eslint-plugin-perfectionist/issues/520.

Here is the generation in this case:

```ts
  /**
   * Enforce sorted imports.
   * @see https://perfectionist.dev/rules/sort-imports
   */
  'perfectionist/sort-imports'?: Linter.RuleEntry<PerfectionistSortImports>

[...]

// ----- perfectionist/sort-imports -----
type perfectionist/sort-imports = unknown // Instead of PerfectionistSortImports
```

A minimal reproduction repository is available here: https://github.com/hugop95/eslint-typegen-fallback-bug.

I might find the time to directly make a PR fix this week if this is fine for you. 

### Reproduction

https://github.com/hugop95/eslint-typegen-fallback-bug

### System Info

```Shell
System:
    OS: macOS 14.4.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 82.52 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.12.0 - ~/.nvm/versions/node/v22.12.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.12.0/bin/yarn
    npm: 10.9.0 - ~/.nvm/versions/node/v22.12.0/bin/npm
    pnpm: 9.15.4 - ~/.nvm/versions/node/v22.12.0/bin/pnpm
    bun: 1.1.36 - /opt/homebrew/bin/bun
```

### Used Package Manager

pnpm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [x] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [x] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by antfu** [comment]
Would appreciate the PR! Thanks!
